simulator/modules/prediction/prediction.cpp

```diff
@@ -0,0 +1,83 @@
+/**
+ * rediction.cpp - Simulation of Branch Mispredict Detection stage
+ * Copyright 2015-2018 MIPT-MIPS
+ */
+
+#include "prediction.h"
+
+static constexpr const uint32 FLUSHED_STAGES_NUM = 4;
+
+template <typename ISA>
+Prediction<ISA>::Prediction( bool log) : Log( log)
+{
+    wp_flush_all = make_write_port<bool>("MEMORY_2_ALL_FLUSH", PORT_BW, FLUSHED_STAGES_NUM);
```

Comment:
`MEMORY_2_ALL` is an obsolete name now — please use BRANCH_2_ALL, same for all others